### DISCUSSION TITLE
Dear programmer, goCD I hope to add the following functions

### DISCUSSION THREAD

**Discussion by liuwenzhi0318** [gocd/gocd#10412] [Ideas]
1、Add judgment function between different pipelines or stages
The sketch is as follows
When manually, you can choose which pipeline to trigger

![image](https://user-images.githubusercontent.com/37536326/167568241-a2584a20-6883-4106-93d2-2466b9fe5040.png)

2、There is also a question, how the variables generated by the upstream pipeline are passed to the pipeline below

**Comment by liuwenzhi0318** [comment]
That is, the child pipeline to get the variables of the parent pipeline

**Reply by chadlwilson** [reply]
I don't think I really understand how this is going to work.

Is the value of `a` determined by Pipeline1/stage3? If so, then how can its value be different in the three downstream pipelines?

I don't know if it'll do what you want, but in general the way to pass context between pipelines with the current GoCD is for a job to an earlier pipeline (e.g Pipeline1) to upload an artifact and then the relevant jobs in downstream pipelines (Pipeline2/3/4) to fetch that artifact. If it is a set of environment variables it could just be an env file that can be sourced or loaded into the shell with a script task.

**Reply by liuwenzhi0318** [reply]
thank you for your reply。The variable a is Pipeline1/stage3 automatically generated，The variable a is a value I get after running, and it will generate several different values. I just need different values ​​to perform different steps later.

**Reply by chadlwilson** [reply]
Sorry, I'm not sure I understand what you are trying to do or how to best model it with GoCD.

If you always want to run all 3 pipelines for every completion of pipeline1 then you're best to use 3 different variables, or publish an artifact that has all 3 values e.g

```
Pipeline2_a=123
Pipeline3_a=456
Pipeline4_a=789
```
Then the first task in each pipelines job can `fetch` the artifact and pick which variable to use based on convention using the environment variable GoCD sets for the pipeline name.

But not really sure that's what you want to happen.
